src/lib/ui/widgets/CalendarWidget.svelte

```diff
@@ -1,7 +1,129 @@
 <script lang="ts">
 	import type { CalendarWidget } from '$lib/data/widgets';
+	import { json } from '@sveltejs/kit';
 
 	export let widget: CalendarWidget;
+
+	// Initial values for the current date and time
+	let newDate = new Date();
+	let currentdate = String;
+	let currenttime = String;
+
+	// Function to display the current date and time
+	function displayDateTime() {
+		newDate = new Date();
+		currentdate = `${newDate.getDate()}.${newDate.getMonth() + 1}.${newDate.getFullYear()}`;
+		currenttime = `${newDate.getHours()}:${newDate.getMinutes()} Uhr`;
+	}
+
+	displayDateTime();
+
+	// Reload date and time every minute
+	setInterval(() => {
+		displayDateTime();
+	}, 60000);
+
+	// call the api with the year and loacation
+	const year = newDate.getFullYear();
+	//const nextYear = newDate.getFullYear() + 1;
+
+	const getHolidays = async () => {
+		const response = await fetch('https://date.nager.at/api/v3/publicholidays/' + year + '/DE');
+		const holidaysForThisYear = await response.json();
+		const holidays = [];
+
+		for (let i = 0; i < holidaysForThisYear.length; i++) {
+			if (
+				compareDates(currentdate, formatDate(holidaysForThisYear[i].date)) < 0 &&
+				incresment() <= 3
+			) {
+				holidays.push(holidaysForThisYear[i]);
+			}
+			if (holidays.length < 3) {
+				const response2 = await fetch(
```

Comment:
Verschiebe diesen Teil lieber aus der For-Schleife raus und füge so viele Feiertage aus dem nächsten Jahr in das Array ein, wie noch Platz ist (also 3 minus die aktuelle Anzahl an Feiertagen).

Wenn im Jahr nur noch ein Feiertag übrig ist, dann würde diese Schleife hier 2 mal laufen und auch 2 mal eine Anfrage an die API stellen, was wir gar nicht müssen